apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/SignedUrlResumableUpload.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            return new SignedUrlResumableUpload(
+                GaxPreconditions.CheckNotNull(signedUrl, nameof(signedUrl)),
+                contentStream,
+                options);
+        }
+
+        /// <inheritdoc/>
+        protected override async Task<Uri> InitiateSessionAsync(CancellationToken cancellationToken)
+        {
+            var httpClient = Options?.HttpClient ?? new HttpClient();
+            var request = new HttpRequestMessage(HttpMethod.Post, SignedUrl);
+            request.Headers.Add("x-goog-resumable", "start");
+            Options?.ModifySessionInitiationRequest?.Invoke(request);
+            var result = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+            if (!result.IsSuccessStatusCode)
+            {
+                throw await ExceptionForResponseAsync(result).ConfigureAwait(false);
+            }
+            return result.Headers.Location;
+        }
+
+        /// <summary>
+        /// Initiates the resumable upload session by posting to the signed URL and returns the session URI.
+        /// </summary>
+        /// <param name="signedUrl">
+        /// The signed URL which can be used to initiate a resumable upload session. See
+        /// <see cref="UrlSigner.ResumableHttpMethod">UrlSigner.ResumableHttpMethod</see> for more information.
+        /// </param>
+        /// <param name="options">The options for the upload operation.</param>
+        /// <returns>
+        /// The session URI to use for the resumable upload.
+        /// </returns>
+        public static Uri InitiateSession(string signedUrl, ResumableUploadOptions options = null) =>
+            GetResult(token => InitiateSessionAsync(signedUrl, options));
+
+        /// <summary>
+        /// Initiates the resumable upload session by posting to the signed URL and returns the session URI.
+        /// </summary>
+        /// <param name="signedUrl">
+        /// The signed URL which can be used to initiate a resumable upload session. See
+        /// <see cref="UrlSigner.ResumableHttpMethod">UrlSigner.ResumableHttpMethod</see> for more information.
+        /// </param>
+        /// <param name="options">The options for the upload operation.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>
+        /// A task containing the session URI to use for the resumable upload.
+        /// </returns>
+        public static Task<Uri> InitiateSessionAsync(
+            string signedUrl,
+            ResumableUploadOptions options = null,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var uploader = new SignedUrlResumableUpload(signedUrl, new MemoryStream(), options);
+            return uploader.InitiateSessionAsync(cancellationToken);
+        }
+
+        private static T GetResult<T>(Func<CancellationToken, Task<T>> operation)
```

Comment:
We now have a method in GAX for the unwrapping part:
https://github.com/googleapis/gax-dotnet/blob/master/src/Google.Api.Gax/TaskExtensions.cs#L37

So you can delete this method and just use

```csharp
InitiateSession(string signedUrl, ResumableUploadOptions options = null) =>
    InitiateSessionAsync(signedUrl, options).ResultWithUnwrappedExceptions();
```